operations/_scripts/generate/ansible/generate_st2_conf.sh

```diff
@@ -0,0 +1,9 @@
+#! /bin/bash
+echo "In generate_st2_conf.sh"
+
+if [[ -n $ST2_CONF_PATH ]]; then
+    cp $GITHUB_WORKSPACE/$ST2_CONF_PATH $GITHUB_ACTION_PATH/operations/deployment/ansible/vars/
+fi
+
+# Delete the default?
+rm $GITHUB_ACTION_PATH/operations/deployment/ansible/vars/st2.yaml
```

Comment:
Indeed deleting might lead to issues, because it relies on some configurations from the user's inputs in the Action.

We should benefit from the Ansible var precedence (https://github.com/stackStorm/ansible-st2#variables) because under the hood the vars are merged.

Where the user-specified vars file should be placed to make sure it'll have more priority over the defaults file?

Reply:
```
command line values (for example, -u my_user, these are not variables)
role defaults (defined in role/defaults/main.yml) [1](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_variables.html#id14)
 inventory file or script group vars [2](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_variables.html#id15)
 inventory group_vars/all [3](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_variables.html#id16)
playbook group_vars/all [3](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_variables.html#id16)
inventory group_vars/* [3](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_variables.html#id16)
 playbook group_vars/* [3](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_variables.html#id16)
 inventory file or script host vars [2](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_variables.html#id15)
 inventory host_vars/* [3](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_variables.html#id16)
playbook host_vars/* [3](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_variables.html#id16)
host facts / cached set_facts [4](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_variables.html#id17)
play vars
play vars_prompt
play vars_files
role vars (defined in role/vars/main.yml)
block vars (only for tasks in block)
task vars (only for the task)
include_vars           <--- This is the singular step you're referring to 
set_facts / registered vars
role (and include_role) params
include params
extra vars (for example, -e "user=my_user")(always win precedence)
        ```
        
        You're putting A LOT of trust in a single merge step in Ansible. One which doesn't understand what the meaning of "default" is.

Reply:
command line values (for example, -u my_user, these are not variables)
role defaults (defined in role/defaults/main.yml) [1](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_variables.html#id14)
 inventory file or script group vars [2](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_variables.html#id15)
 inventory group_vars/all [3](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_variables.html#id16)
playbook group_vars/all [3](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_variables.html#id16)
inventory group_vars/* [3](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_variables.html#id16)
 playbook group_vars/* [3](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_variables.html#id16)
 inventory file or script host vars [2](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_variables.html#id15)
 inventory host_vars/* [3](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_variables.html#id16)
playbook host_vars/* [3](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_variables.html#id16)
host facts / cached set_facts [4](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_variables.html#id17)
play vars
play vars_prompt
play vars_files
role vars (defined in role/vars/main.yml)
block vars (only for tasks in block)
task vars (only for the task)
include_vars           <--- This is the singular step you're referring to 
set_facts / registered vars
role (and include_role) params
include params
extra vars (for example, -e "user=my_user")(always win precedence)
        ```
        
        You're putting A LOT of trust in a single merge step in Ansible. One which doesn't understand what the meaning of "default" is.